cvat/apps/webhooks/signals.py

```diff
@@ -136,25 +136,32 @@ def get_sender(instance):
 @receiver(pre_save, sender=Invitation, dispatch_uid=__name__ + ":invitation:pre_save")
 @receiver(pre_save, sender=Membership, dispatch_uid=__name__ + ":membership:pre_save")
 def pre_save_resource_event(sender, instance, **kwargs):
-    try:
-        old_instance = sender.objects.get(pk=instance.pk)
-    except ObjectDoesNotExist:
-        return
+    instance._webhooks_selected_webhooks = []
 
-    old_serializer = get_serializer(instance=old_instance)
-    serializer = get_serializer(instance=instance)
-    diff = get_instance_diff(old_data=old_serializer.data, data=serializer.data)
+    if instance.pk is None:
+        created = True
+    else:
+        try:
+            old_instance = sender.objects.get(pk=instance.pk)
+            created = False
+        except ObjectDoesNotExist:
+            created = True
```

Comment:
Why do we need to handle it like this? If `instance.pk is not None` but the instance does not exist anymore, it means that the object was deleted before this code was executed. After initializing `created = True` the code below will check the wrong event "created:object". IMHO, we should stop handling here (return from function).

Reply:
> If `instance.pk is not None` but the instance does not exist anymore,

I wasn't thinking of this case, but rather of the case where an object is created with a preset public key value. I don't think we _do_ that anywhere right now, but I figured the code should be robust and handle that situation.

But also, in the case you describe, I don't see how my code is incorrect? If the instance was deleted prior the call to `save`, then `.save` will re-create it, so `created:*` seems like the correct event.